Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 F. Nuccio, H. Zheng, W. He, S. Wu, Y. Yuan, W. Jiao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio, Huanyu Zheng, Sihan Wu, Wanyi He, Weichen Jiao, Yi Yuan
+-/
+import Mathlib.Algebra.CharP.LinearMaps
+import Mathlib.Algebra.Field.Power
+import Mathlib.FieldTheory.PurelyInseparable
+
+/-!
+# The Jacobson-Noether theorem
+
+This file contains a proof of the Jacobson-Noether theorem and some auxiliary lemmas.
+Here we discuss different cases of characteristics of
+the noncommutative division algebra `D` with center `k`.
+
+## Main Results
+
+- `Jacobson_Noether` : Let `D` be a noncommutative division algebra with center `k`.
+  If `D` is algebraic over `k`,
+  then there exists an element of `D \ k` which is separable over `k`.
+
+## Notations
+
+- `D` is a noncommutative division algebra
+- `k` is the center of `D`
+
+## Implementation Notes
+
+## Reference
+* <https://ysharifi.wordpress.com/2011/09/30/the-jacobson-noether-theorem/>
+-/
+
+namespace JacobsonNoether
+
+variable {D : Type*} [DivisionRing D]
+
+open LinearMap
+
+local notation3 "k" => (Subring.center D)
+private def δ : D → D →ₗ[k] D := mulLeft k - mulRight k
+
+private lemma δ_def (a x : D) : δ a x = mulLeft k a x - mulRight k a x := rfl
+
+private lemma δ_def' (a : D) : δ a = mulLeft k a - mulRight k a := rfl
+
+private lemma l_pow (a : D) (n : ℕ) : ∀ x : D, ((mulLeft k a) ^ n).1 x = (a ^ n) * x := by
+  intro x
+  rw [coe_toAddHom, pow_apply]
+  induction n with
+  | zero => rw [Function.iterate_zero, id_eq, pow_zero, one_mul]
+  | succ _ h => rw [Function.iterate_succ', Function.comp_apply, h,
+    pow_succ', mul_assoc, mulLeft_apply]
+
+private lemma r_pow (a : D) (n : ℕ) : ∀ x : D, ((mulRight k a) ^ n).1 x = x * (a ^ n) := by
+  intro x
+  rw [coe_toAddHom, pow_apply]
+  induction n with
+  | zero => rw [Function.iterate_zero, id_eq, pow_zero, mul_one]
+  | succ _ h => rw [Function.iterate_succ', Function.comp_apply, h,
+    pow_add, pow_one, mulRight_apply, mul_assoc]
+
+private lemma δ_iterate_succ (x y : D) (n : ℕ) :
+    δ x (((δ x) ^ n) y) = ((δ x) ^ (n + 1)) y := by
+  simp only [pow_apply, δ_def, mulLeft_apply, mulRight_apply, Function.iterate_succ_apply']
+
+variable [Algebra.IsAlgebraic (Subring.center D) D]
+
+open Polynomial
+
+/-- If `D` is a purely inseparable extension of `k` with characteristic `p`,
+  then for every element `a` of `D`, there exists a natural number `n`
+  such that `a ^ (p ^ n)` is contained in `k`. -/
+lemma exists_pow_mem_center_ofInseparable (p : ℕ) [hchar : ExpChar D p] (a : D)
+    (hinsep : ∀ x : D, IsSeparable k x → x ∈ k) : ∃ n, a ^ (p ^ n) ∈ k := by
+  have := (@isPurelyInseparable_iff_pow_mem k D _ _ _ _ p (ExpChar.center_expChar_iff.1 hchar)).1
+  have bridge : ∀ x : k, (algebraMap k D) x = x := fun _ ↦ rfl
```

Comment:
you will only need this once; so instead of adding it here, you can use as a `rw` at `pure` later on (or use the `▸`)